File: src/main/java/com/github/mutl3y/ansiblehelper/AnsibleContext.java
// Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.

package com.github.mutl3y.ansiblehelper;

import com.intellij.codeInsight.template.TemplateActionContext;
import com.intellij.codeInsight.template.TemplateContextType;
import org.jetbrains.annotations.NotNull;

public class AnsibleContext extends TemplateContextType {

  protected AnsibleContext() {
    super("ANSIBLE", "Ansible");
  }

  @Override
  public boolean isInContext(@NotNull TemplateActionContext templateActionContext) {
    return templateActionContext.getFile().getName().endsWith(".yml");
  }

}

File: src/main/kotlin/com/github/mutl3y/ansiblehelper/AnsibleContext.kt
// Copyright 2000-2020 JetBrains s.r.o. and other contributors.
// Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package com.github.mutl3y.ansiblehelper

import com.intellij.codeInsight.template.TemplateContextType
import com.intellij.codeInsight.template.TemplateActionContext

class AnsibleContext private constructor() : TemplateContextType("ANSIBLE", "Ansible") {
    override fun isInContext(templateActionContext: TemplateActionContext): Boolean {
        return templateActionContext.file.name.endsWith(".yml")
    }
}
